packages/labs/analyzer/src/lib/javascript/mixins.ts

```diff
@@ -0,0 +1,201 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * @fileoverview
+ *
+ * Utilities for working with mixins
+ */
+
+import ts from 'typescript';
+import {AnalyzerInterface, MixinDeclaration} from '../model.js';
+import {getClassDeclaration} from './classes.js';
+import {createDiagnostic} from '../errors.js';
+import {DiagnosticCode} from '../diagnostic-code.js';
+import {parseNodeJSDocInfo} from './jsdoc.js';
+import {getFunctionLikeInfo} from './functions.js';
```

Comment:
FYI this introduces a circular dependency as `functions.js` imports `mixins.js`

not too sure what you want to do about that